ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx

```diff
@@ -16,95 +34,61 @@ type CLIDownloadMenuProps = {
-        </ApplicationLauncherItem>,
-        <ApplicationLauncherItem
-            key="app-launcher-item-cli-linux-arm64"
-            component="button"
-            onClick={handleDownloadCLI('linux-arm64')}
-        >
-            Linux arm_64
-        </ApplicationLauncherItem>,
-        <ApplicationLauncherItem
-            key="app-launcher-item-cli-linux-ppc64le"
-            component="button"
-            onClick={handleDownloadCLI('linux-ppc64le')}
-        >
-            Linux ppc64le
-        </ApplicationLauncherItem>,
-        <ApplicationLauncherItem
-            key="app-launcher-item-cli-linux-s390x"
-            component="button"
-            onClick={handleDownloadCLI('linux-s390x')}
-        >
-            Linux s390x
-        </ApplicationLauncherItem>,
-        <ApplicationLauncherItem
-            key="app-launcher-item-cli-windows-amd64"
-            component="button"
-            onClick={handleDownloadCLI('windows-amd64')}
-        >
-            Windows x86_64
-        </ApplicationLauncherItem>,
-    ];
-
-    function toggleCLIMenu() {
-        setIsCLIMenuOpen(!isCLIMenuOpen);
-    }
-
-    const CLIDownloadIcon = (
-        <Flex alignItems={{ default: 'alignItemsCenter' }} spaceItems={{ default: 'spaceItemsSm' }}>
-            <FlexItem>
-                <DownloadIcon alt="" />
-            </FlexItem>
-            <FlexItem>CLI</FlexItem>
-        </Flex>
-    );
-
     return (
-        <ApplicationLauncher
-            aria-label="CLI Download Menu"
-            onToggle={toggleCLIMenu}
+        <Dropdown
             isOpen={isCLIMenuOpen}
-            items={appLauncherItems}
-            position="right"
-            data-quickstart-id="qs-masthead-climenu"
-            toggleIcon={CLIDownloadIcon}
-        />
+            onOpenChange={(isOpen) => setIsCLIMenuOpen(isOpen)}
+            onOpenChangeKeys={['Escape', 'Tab']}
+            popperProps={{ position: 'right' }}
+            toggle={(toggleRef) => (
+                <MenuToggle
+                    aria-label="CLI Download Menu"
+                    ref={toggleRef}
+                    variant="plain"
+                    onClick={() => setIsCLIMenuOpen((wasOpen) => !wasOpen)}
+                    isExpanded={isCLIMenuOpen}
+                >
+                    <Flex
+                        alignItems={{ default: 'alignItemsCenter' }}
+                        spaceItems={{ default: 'spaceItemsSm' }}
+                    >
+                        <FlexItem>
+                            <DownloadIcon alt="Download roxctl" />
+                        </FlexItem>
+                        <FlexItem>CLI</FlexItem>
+                    </Flex>
+                </MenuToggle>
+            )}
+        >
+            <DropdownList>
+                {cliDownloadOptions.map(({ os, display }) => (
+                    <DropdownItem
+                        id={`app-launcher-item-cli-${os}`}
+                        value={`app-launcher-item-cli-${os}`}
```

Comment:
First, it seems safest for `onClick` handler to use `os` argument, as usual.

For my understanding, are `id` and `value` props needed? Here is blurb for `value` prop:

https://www.patternfly.org/components/menus/dropdown#dropdownitem
> Identifies the component in the dropdown onSelect callback

Reply:
Good call, those are left over from the old implementation. I'll go ahead and remove them as they do not cause a change in behavior or aXe findings.